x-pack/platform/packages/shared/kbn-inference-tracing/src/phoenix/phoenix_span_processor.ts

```diff
@@ -80,9 +113,10 @@ export class PhoenixSpanProcessor extends BaseInferenceSpanProcessor {
           return;
         }
 
-        const url = new URL(
-          `/projects/${projectId}/traces/${traceId}?selected`,
-          new URL(this.config.public_url)
+        const publicBase = getDirectoryBaseUrl(new URL(this.config.public_url));
+        const url = resolvePathWithinBasePrefix(
+          publicBase,
+          `/projects/${projectId}/traces/${traceId}?selected`
```

Comment:
What about something like:

```ts
function getPhoenixUrl(base: string, path: string) {
  const baseWithTrailingSlash = base.endsWith('/') ? base : `${base}/`;
  const pathWithoutLeadingSlash = path.startsWith('/') ? path.slice(1) : path;
  return new URL(pathWithoutLeadingSlash, baseWithTrailingSlash);
}
```

It can be invoked like:

```ts
getPhoenixUrl(this.config.public_url, `/projects/${projectId}/traces/${traceId}?selected`);
```

and

```ts
getPhoenixUrl(config.base_url, '/v1/projects')
```